modules/platform/nuxeo-platform-web-common/src/test/java/org/nuxeo/ecm/platform/web/requestcontroller/filter/TestIdempotentRequestFilter.java

```diff
@@ -0,0 +1,345 @@
+        }
+
+    }
+
+    @Before
+    public void setUp() throws IOException {
+        filter = new NuxeoRequestControllerFilter();
+        FilterConfig config = mock(FilterConfig.class);
+        when(config.getInitParameter(IdempotentRequestHandler.TTL_SECONDS_PROPERTY)).thenReturn(String.valueOf(TTL));
+        when(config.getInitParameter(IdempotentRequestHandler.STORE_PROPERTY)).thenReturn(KV_STORE_NAME);
+        filter.init(config);
+        chain = mock(FilterChain.class);
+        request = mock(HttpServletRequest.class);
+        mockResponse = new MockResponse();
+        // handle store
+        store = (KeyValueStoreProvider) kvs.getKeyValueStore(KV_STORE_NAME);
+        store.clear();
+    }
+
+    @After
+    public void tearDown() {
+        if (filter != null) {
+            filter.destroy();
+        }
+        if (store != null) {
+            store.clear();
+        }
+    }
+
+    protected void checkResponse(MockResponse mockResponse, Integer status, String content) {
+        assertEquals(status, (Integer) mockResponse.getStatus());
+        assertEquals(content, mockResponse.getOutput().toString());
+    }
+
+    protected void checkStore(Integer status, String content) {
+        assertEquals(content, store.getString(IdempotentRequestHandler.CONTENT_PREFIX + KEY));
+        String skey = IdempotentRequestHandler.STATUS_PREFIX + KEY;
+        if (status == null) {
+            assertNull(store.getString(skey));
+        } else {
+            assertEquals(String.valueOf(status), store.getString(skey));
+        }
+    }
+
+    @Test
+    public void testGetRequestWithoutKey() throws Exception {
+        when(request.getMethod()).thenReturn("GET");
+        verify(chain, times(0)).doFilter(any(), any());
+        filter.doFilter(request, mockResponse.getResponse(), chain);
+        verify(chain, times(1)).doFilter(any(), any());
+        checkStore(null, null);
+    }
+
+    @Test
+    public void testPostRequestWithoutKey() throws Exception {
+        when(request.getMethod()).thenReturn("POST");
+        verify(chain, times(0)).doFilter(any(), any());
+        filter.doFilter(request, mockResponse.getResponse(), chain);
+        verify(chain, times(1)).doFilter(any(), any());
+        checkStore(null, null);
+    }
+
+    @Test
+    public void testGetRequest() throws Exception {
+        when(request.getMethod()).thenReturn("GET");
+        when(request.getHeader(IdempotentRequestHandler.HEADER_KEY)).thenReturn(KEY);
+        verify(chain, times(0)).doFilter(any(), any());
+        filter.doFilter(request, mockResponse.getResponse(), chain);
+        verify(chain, times(1)).doFilter(any(), any());
+        checkStore(null, null);
+    }
+
+    protected void setResult(HttpServletResponse response, int status, String content) throws IOException {
+        response.setStatus(status);
+        response.getWriter().write(content);
+    }
+
+    @Test
+    public void testPostRequest() throws Exception {
```

Comment:
In addition to this base test we should test for content that is not representable as a string, and headers (content type, custom headers, cookies).

Reply:
I added quick mock headers, as for the content, not sure if this would belong in this test as everything here is actually mocked... to be discussed again, keeping this conversation open for this.